dao/src/main/java/org/thingsboard/server/dao/service/validator/DeviceProfileDataValidator.java

```diff
@@ -210,6 +209,16 @@ protected void validateDataImpl(TenantId tenantId, DeviceProfile deviceProfile)
         validateOtaPackage(tenantId, deviceProfile, deviceProfile.getId());
     }
 
+    private void validateRuleChain(TenantId tenantId, TenantId deviceProfileTenantId, RuleChainId ruleChainId) {
```

Comment:
What is the difference between deviceProfileTenantId and tenantId? Make it one parameter